### PR TITLE
feat: icky 893 search and filter account list

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/dialogs/b2b/B2bContactsFilterDialog/B2bContactsFilterDialog.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+
+import B2bContactsFilterDialog from './B2bContactsFilterDialog'
+
+export default {
+  title: 'Dialogs/B2B/B2bContactsFilterDialog',
```
Comment (Kibo-TriptiV) [inline components/dialogs/b2b/B2bContactsFilterDialog/B2bContactsFilterDialog.stories.tsx]: we can keep it on common dialog instead inside b2b. Lets discuss tomorrow. ---> , As discussed, dialog is in correct place. Please rename the file name to AddressContactFilter dialog.

**Review by Kibo-TriptiV — COMMENTED** [review]
overall looks good. lets discuss tomorrow regarding comment.

File: components/b2b/seller/AccountsTable/AccountsTable.tsx [hunk 2]
```diff
@@ -19,10 +19,15 @@ import { useTranslation } from 'next-i18next'
...
 import { AccountsTableStyles } from './AccountsTable.styles'
 import { KiboPagination, SearchBar } from '@/components/common'
+import { B2bContactsFilterDialog } from '@/components/dialogs'
+import { useModalContext } from '@/context'
 import { useDebounce } from '@/hooks'
 import { addressGetters } from '@/lib/getters'
+import { buildB2bContactFilterParam } from '@/lib/helpers'
+import { B2bContactsFilters } from '@/lib/types'
+
+import { QueryQuotesArgs } from '@/lib/gql/types'
```
Comment (Kibo-TriptiV) [inline components/b2b/seller/AccountsTable/AccountsTable.tsx]: use type everywhere

File: components/b2b/seller/AccountsTable/AccountsTable.tsx [hunk 3]
```diff
@@ -78,20 +84,19 @@ const AccountsTable = (props: AccountsTableProps) => {
...
-  // const handleFilterButtonClick = () => {
-  //   showModal({
-  //     Component: QuotesFilterDialog,
-  //     props: {
-  //       filters: filters,
-  //       // onFilterAction: handleFilterAction,
-  //     },
-  //   })
-  // }
+  const handleFilterAction = (filters: B2bContactsFilters) => {
```
Comment (Kibo-TriptiV) [inline components/b2b/seller/AccountsTable/AccountsTable.tsx:86]: Please write the test cases for filter and search functionality added.

**Review by Kibo-TriptiV — CHANGES_REQUESTED** [review]
Please check review comments.

File: components/b2b/seller/AccountsTable/AccountsTable.tsx (hunk 3 above)
Comment (kibo-chandradeeptalaha) [reply components/b2b/seller/AccountsTable/AccountsTable.tsx:86]: Added

**Review by Kibo-TriptiV — APPROVED** [review]

**Pull request merged**